src/interfaces.ts

```diff
@@ -201,14 +217,62 @@ interface PipelineStep {
    * @param next The next step in the pipeline.
    */
   run(instruction: NavigationInstruction, next: Next): Promise<any>;
+
+  /**
+   * @internal
+   */
+  getSteps?(): any[];
 }
 
 /**
 * The result of a pipeline run.
 */
-interface PipelineResult {
+export interface PipelineResult {
   status: string;
   instruction: NavigationInstruction;
   output: any;
   completed: boolean;
 }
+
+/**
+ * The component responsible for routing
+ */
+export interface RoutingComponent {
+  viewModel: any;
+  childContainer?: Container;
+  router?: Router;
+  config?: RouteConfig;
+  childRouter?: Router;
+}
+
+export interface Viewport {
+  /**@internal */
+  container: Container;
+  swap(viewportInstruction: ViewPortInstruction): void;
+  process(viewportInstruction: ViewPortInstruction, waitToSwap?: boolean): void;
+}
+
+export interface ViewPortInstruction {
```

Comment:
Having name be the only required part doesn't make sense to me. I don't think name is used anywhere at all. I believe strategy, config, moduleId, component, and lifecycleArgs are all required.

Reply:
I needed to make it partial to get it to work as easily as possible. Will improve the typing here